src/IPAddr.h

```diff
@@ -53,6 +53,8 @@ class ConnKey {
     void Init(const IPAddr& src, const IPAddr& dst, uint16_t src_port, uint16_t dst_port, uint16_t proto, bool one_way);
 };
 
+using ConnKeyPtr = std::shared_ptr<ConnKey>;
```

Comment:
Complete drive-by comment: It looks like we never intentionally create shared copies of values of this type, but we always incur the overhead of updating the ref count. I believe this should be

```cpp
using ConnKeyPtr = std::shared_ptr<ConnKeyPtr>;
```

Doing that requires cleaning up some functions which currently take `const ConnKeyPtr` (an immutable value, not a ref) where they should just take full ownership by passing `ConnKeyPtr` if they consume the value, or `const ConnKeyPtr` if they just access it.

Reply:
> using ConnKeyPtr = std::shared_ptr<ConnKeyPtr>;

I'm confused here. Isn't that creating a shared_ptr to the same aliased type?

Reply:
I updated the suggestion, this should have just been a `unique_ptr` if at all.